core/controllers/resources.py

```diff
@@ -48,18 +48,18 @@ class ImageHandler(base.BaseHandler):
     _IMAGE_PATH_PREFIX = 'image'
 
     @acl_decorators.open_access
-    def get(self, exploration_id, encoded_filepath):
+    def get(self, exploration_id, filename):
         """Returns an image.
 
         Args:
             exploration_id: the id of the exploration.
-            encoded_filepath: a string representing the image filepath. This
+            filename: a string representing the image filepath. This
               string is encoded in the frontend using encodeURIComponent().
         """
         if not constants.DEV_MODE:
             raise self.PageNotFoundException
         try:
-            filepath = urllib.unquote(encoded_filepath)
+            filepath = urllib.unquote(filename)
```

Comment:
That makes sense. I think the code for audio handler below is similar. Do we need quote escaping there as well?

Reply:
I think so (and I think you've added it!) But the filepath thing still needs changing, I think.

It would also be nice to unify these handlers into a single AssetDevHandler, too, IMO (again to mimic GCS better). But that can happen in the future, in a separate PR.

Reply:
Thanks Sean, I made the filepath change and filed #5757 to unify asset handlers.